sprite.cpp

```diff
@@ -52,9 +52,9 @@ void Sprite::update(Context &ctx) {
 		}
 	};
 
-	float edge_boundary = 0.15f + m_size / 1.1f;
-	float horizontal_correction = max((float) ctx.rect().right / (float) ctx.rect().bottom, 1.0f);
-	float vertical_correction = max((float) ctx.rect().bottom / (float) ctx.rect().right, 1.0f);
+	double edge_boundary = 0.15f + m_size / 1.1f;
+	double horizontal_correction = max((double) ctx.rect().right / (double) ctx.rect().bottom, 1.0f);
+	double vertical_correction = max((double) ctx.rect().bottom / (double) ctx.rect().right, 1.0f);
```

Comment:
I don't know if this is intentional or a mistake, but you are still using floats here